warehouse/accounts/security_policy.py

```diff
@@ -102,7 +102,7 @@ def identity(self, request):
             return None
 
         # Sessions can only authenticate users, not any other type of identity.
-        return user
+        return UserContext(user, None)
```

Comment:
```suggestion
        return UserContext(user=user, macaroon=None)
```
Just to add a bit of clarity.